merge.cpp

```diff
@@ -53,46 +52,101 @@ void PrintSingularValues(const int rank, const std::string& name, CAROM::SVDBasi
 }
 
 void LoadSampleSets(const int rank, const double energyFraction, const int nsets, const std::string& varName,
-                    const int dim, const int totalSamples)
+                    const bool usingWindows, const int window, const int dim, const int totalSamples, int& cutoff)
 {
     std::unique_ptr<CAROM::SVDBasisGenerator> basis_generator;
 
-    std::string basis_filename = "run/basis" + varName + "0";  // 0 is the time window index
+    std::string basis_filename = "run/basis" + varName + std::to_string(window);
     basis_generator.reset(new CAROM::StaticSVDBasisGenerator(dim, totalSamples, basis_filename));
 
-    cout << "Loading snapshots for " << varName << endl;
+    cout << "Loading snapshots for " << varName << " in time window " << window << endl;
 
     for (int i=0; i<nsets; ++i)
     {
-        std::string filename = "run/param" + std::to_string(i) + "_var" + varName + "0_snapshot";  // 0 is time window index
+        std::string filename = "run/param" + std::to_string(i) + "_var" + varName + std::to_string(window) + "_snapshot";
         basis_generator->loadSamples(filename,"snapshot");
     }
 
     //cout << "Saving data uploaded as a snapshot" << endl;
     //basis_generator->writeSnapshot();
 
-    cout << "Computing SVD for " << varName << endl;
+    cout << "Computing SVD for " << varName << " in time window " << window << endl;
     basis_generator->endSamples();  // save the basis file
 
     if (rank == 0)
     {
         cout << varName << " basis summary output: ";
-        BasisGeneratorFinalSummary(basis_generator.get(), energyFraction);
-        PrintSingularValues(rank, varName, basis_generator.get());
+        BasisGeneratorFinalSummary(basis_generator.get(), energyFraction, cutoff);
+        PrintSingularValues(rank, varName, usingWindows, window, basis_generator.get());
     }
 }
 
 // id is snapshot index, 0-based
-void GetSnapshotDim(const int id, const std::string& varName, int& varDim, int& numSnapshots)
+void GetSnapshotDim(const int id, const std::string& varName, const int window, int& varDim, int& numSnapshots)
 {
-    std::string filename = "run/param" + std::to_string(id) + "_var" + varName + "0_snapshot";  // 0 is time window index
+    std::string filename = "run/param" + std::to_string(id) + "_var" + varName + std::to_string(window) + "_snapshot";
 
     CAROM::BasisReader reader(filename);
     const CAROM::Matrix *S = reader.getSnapshotMatrix(0.0);
     varDim = S->numRows();
     numSnapshots = S->numColumns();
 }
 
+int ReadTimeWindows(const int nw, std::string twfile, Array<double>& twep, const bool printStatus)
+{
```

Comment:
@dylan-copeland @siuwuncheung Maybe not in this PR, but what do we think about moving ReadTimeWindows, PrintSingularValues, BasisGeneratorFinalSummary, and some of the other functions that don't belong in a class into a utils file so we don't have to repeat the function definition twice in the various Laghos files and merge.cpp?